apps/prairielearn/python/check_data.py

```diff
@@ -0,0 +1,168 @@
+from typing import Any, Dict, List, Literal, TypedDict
+
+Phase = Literal["generate", "prepare", "render", "parse", "grade", "test", "file"]
+
+ValueType = Literal["boolean", "integer", "number", "string", "object"]
+
+all_phases: List[Phase] = [
+    "generate",
+    "prepare",
+    "render",
+    "parse",
+    "grade",
+    "test",
+    "file",
+]
+
+
+class PropInfo(TypedDict):
+    type: ValueType
+    present_phases: List[Phase]
```

Comment:
Nit: This list is meant to never change and contain unique elements, not likely to be an issue but nice to express this with the data structure choice.

```suggestion
    present_phases: FrozenSet[Phase]
```

Reply:
Done in 8e121e4.